client/src/utils/api.js

```diff
@@ -174,3 +175,110 @@ export const deleteResource = (id: string): ApiResponse<null> => {
       },
     );
 };
+
+export const addCategory = (category: Category): ApiResponse<Category> =>
+  instance
+    .post('/api/admin/categories', category, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+
+export const editCategory = (
+  id: string,
+  category: Category,
+): ApiResponse<Category> => {
+  const requestExtension = `/api/admin/categories/${id}`;
+  return instance
+    .put(requestExtension, category, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+};
+
+export const deleteCategory = (id: string): ApiResponse<null> => {
+  const requestExtension = `/api/admin/categories/${id}`;
+  return instance
+    .delete(requestExtension, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+};
+
+export const addSubcategory = (
+  id: string,
+  category: Category
+): ApiResponse<Category> =>
+  instance
+    .post(`/api/admin/subcategories/${id}`, category, {
+      headers: {
+        token: localStorage.getItem('token'),
+      },
+    })
+    .then(
+      (res) => res.data,
+      (err) => {
+        console.error(err);
+        return null;
+      },
+    );
+
+export const editSubcategory = (
+  id: string,
+  category: Category,
```

Comment:
Similarly to the above comments, in the endpoint we're just using `category`, `newName`, and `currentName`, so we can change the type of our `category` parameter to be a `string`, and then add the `newName` and `currentName` parameters - and then pass all of that into the put by doing `{ category, newName, currentName }`